core/js/typespecs/app_types.ts

```diff
@@ -169,6 +169,27 @@ export const envVarSchema = z
   .openapi("EnvVar");
 export type EnvVar = z.infer<typeof envVarSchema>;
 
+export const customColumnObjectTypeEnum = z
+  .enum(["experiment", "dataset", "project_log"])
+  .describe("The type of the object the custom column is scoped for");
+
+const customColumnBaseSchema = generateBaseTableSchema("custom columns");
+export const customColumnSchema = z
+  .object({
+    id: customColumnBaseSchema.shape.id,
+    object_type: customColumnObjectTypeEnum,
+    object_id: z
+      .string()
+      .uuid()
+      .describe("The id of the object the custom column is scoped for"),
+    name: z.string().describe("The name of the custom column"),
+    value: z.string().describe("The name of the custom column"),
```

Comment:
> value of the custom column

Should this be path or something?

Reply:
yeah good point, I didn't want to necessarily restrict it to paths but can rename to expr